spinalcordtoolbox/cropping.py

```diff
@@ -13,9 +13,7 @@
 
 
 class BoundingBox(object):
-    """
-    """
-    def __init__(self, xmin=None, xmax=None, ymin=None, ymax=None, zmin=None, zmax=None):
+    def __init__(self, xmin, xmax, ymin, ymax, zmin, zmax):
```

Comment:
<details>
<summary> Aside </summary>

I felt that this was a little difficult + time-intensive to review, I think because a few different changes were spread across these two commits, and I had to jump back and forth between the commits to fully understand what had changed.

</details>

Reply:
> Case in point: I should have mentioned that the default values for `arguments.xmin`, `arguments.xmax`, etc. are not `None`, but rather 0 and -1, see [the argument declarations here](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/2df887bfdeee82039964aaa2ae9b4f739a5a5c31/spinalcordtoolbox/scripts/sct_crop_image.py#L71-L85).

Oh!! Of course! Thank you for pointing this out for me. I indeed completely missed that. 😅 

> I did wonder about that, and reasoned that this would match the behaviour at the command line when not passing any arguments, but I agree that raising an explicit error is better. I'll change it.

Now that I understand the default behaviour when not passing any arguments, your original reasoning makes more sense to me! 

But, that makes me wonder... Should we even consider "calling `sct_crop_image` with no arguments" valid usage? 🤔

Reply:
> Sorry for the difficult review. I'd be happy to try out different workflows and find out what's easiest (and part of that is getting feedback about when it's hard, so thanks!)
> 
> For example, it may be easier if I separate the refactoring and the bug-fixing into different PRs (as I did in #3774). I naturally do both at the same time in my process for understanding the code, but I can still draft separate PRs.
> 
> I do think some of the tangled-ness comes from the original code, though, so probably refactoring PRs will be somewhat harder to review in general. I'll try to explicitly state my findings about the code (like invariants, assumptions, etc.), which I didn't do enough here.

Oh gosh, no worries whatsoever. Like you said, a lot of the difficulty was inherent to the... messiness of the existing codebase, hehe. 

In past months, I've tended to just... work around the mess, and only make very surgical, light changes? So, having someone take the time to actually untangle it all is simultaneously a big boon and also a bit of a new challenge, review-wise. But, I think it'll be worth it in the end. :)

Reply:
I think "calling `sct_crop_image` with no cropping-related arguments"  should be fine, since there are documented default values for each of `xmin`, `xmax`, etc. Also, it does emit a log message with the bounding box it's using, so it wouldn't be a hard-to-debug silent problem.

Personally, I like having sensible defaults in my tools. A tool that throws an error when I ask it for a no-op means that it's harder to use in a pipeline or a script.